### PR TITLE
Added github action to draft PR without issue number

### PR DESCRIPTION
**Pull request by kurtisassad**
## Link to Issue
Closes: #4258

## Description of Changes
- Adds CI step which will draft a PR if the issue number is not present. 

## Test Plan
- Created a PR without an issue number, watched this CI step turn it into draft
- Created a PR with an issue number (this one), watched CI step keep it open

### CODE REVIEW

**Review by CowMuon — APPROVED** [review]
LGTM

File: .github/workflows/draft_without_issue.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  checkLinkedIssue:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/github-script@v6
+        id: set-result
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const prDescription = context.payload.pull_request.body;
+            const issueMatcher = /[cC]loses:\s*#(\d+)/;
```
Comment (jnaviask) [inline .github/workflows/draft_without_issue.yml:18]: "Fixes" is also valid -- we shouldn't rely on regex unless we really have no other choice...

**Review by jnaviask — CHANGES_REQUESTED** [review]
Is there any way to query the PR's metadata for a linked issue, rather than looking at the description?

**Review by Rotorsoft — APPROVED** [review]
LGTM 🚀

**Review by CowMuon — APPROVED** [review]
✅

**Review by jnaviask — APPROVED** [review]

**Pull request merged**